src/ckanext-tdc/ckanext/tdc/logic/action.py

```diff
@@ -426,11 +426,47 @@ def _add_display_name_to_multi_select_facets(search_response):
                             item["display_name"] = value_label
 
 
+def _fix_facet_items_per_field_type(result):
+    dataset_schema = scheming_helpers.scheming_get_dataset_schema("dataset")
+    search_facets = result.get("search_facets", [])
+
+    if len(search_facets) > 0:
+        dataset_fields = dataset_schema["dataset_fields"]
+
+        for facet_name in search_facets:
+            facet_field = list(filter(lambda x: x.get(
```

Comment:
Create a dict outside of this scope and run a "for" once to get each face_field and store the facet name as the key of the dict, then inside of facet_name "for" you just try to get from the dict by key.

Code example:

```python
fields_by_key = dict()
for field in dataset_fields:
     fields_by_key[field.get('field_name')] = field
```

Your code is executing O(n * j):
n = search_facets length
j = dataset_fields length

Using the approach above it'll run in O(n + j) which is equals to O(n)

Reply:
Also, the code will be more legible using the approach above.